===== Patch instructions generated by SAVER =====
[+] { Insert: if ({ p11_kit_pin_new:n$8 == { 0 } }) free(*(p11_kit_pin_new:copy)) at 4 (line 554, column 2) }
===== Actual patch in diff format =====
@@ -551,6 +551,8 @@ p11_kit_pin_new (const unsigned char *value, size_t length)
 
 	memcpy (copy, value, length);
 	pin = p11_kit_pin_new_for_buffer (copy, length, free);
+	if (pin == 0)
+		free(copy);
 	return_val_if_fail (pin != NULL, NULL);
 
 	return pin;
